mmdet/utils/deployment/ssd_export_helpers.py

```diff
@@ -173,15 +174,25 @@ def forward(ctx, cls_scores, bbox_preds, img_metas, cfg,
         return output
 
 
-def onnx_export(self, img, img_metas, export_name='', **kwargs):
+def save_img_metas(self, img_metas):
     self._export_mode = True
     self.img_metas = img_metas
+
+
+def onnx_export(self, img, img_metas, export_name='', **kwargs):
+    # Note that this saving img_metas should be done using this method
+    # `save_img_metas` (not using direct access to a field) to be sure
+    # that for NNCF-wrapped models this method is called for original model,
+    # not for NNCF-wrapper of the model.
+    # To make this trick the method `save_img_metas` should be created
+    # (using `obj.save_img_metas = save_img_metas.__get__(obj)`) for the original model.
+    self.save_img_metas(img_metas)
     torch.onnx.export(self, img, export_name, **kwargs)
 
 
 def forward(self, img, img_meta=[None], return_loss=True,
             **kwargs):  # passing None here is a hack to fool the jit engine
-    if self._export_mode:
+    if self._export_mode or kwargs.get('forward_export', False):
```

Comment:
Why `_export_mode` alone is not enough?

Reply:
It's need for wrap_nncf_export function because _export_mode set only in onnx_export.